### DISCUSSION TITLE
Const in match range: support use of const depending on other consts.

### DISCUSSION THREAD

**Discussion by ytoml** [vlang/v#18624] [Ideas]
### Describe the feature

Recently, use of constant values with range in match branch is supported thanks to @Delta456.
(see #16480, #16526)

However, I found that compiler still inhibits const depending on other const in match range, i.e., 
```v
fn can_be_compiled() {
	match 0 {
		a ... b {}
		else {}
	}
}

fn cannot_be_compiled() {
	match 0 {
		a ... c {}
		else {}
	}
}
```
will encounter compile error:
```
main.v:27:9: error: match branch range expressions need the end value to be known at compile time (only enums, const or literals are supported)
   25 | fn cannot_be_compiled() {
   26 |     match 0 {
   27 |         a ... c {}
      |               ^
   28 |         else {}
   29 |     }
```

I'm very happy if this will be supported 🙏

### Use Case

Same as #16480.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.2 21d6bd9

### Environment details (OS name and version, etc.)

OS: macos, macOS, 12.5, 21G72
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)

vexe mtime: 2022-12-25 08:18:29
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 9f1239c.21d6bd9

Git version: git version 2.37.1 (Apple Git-137.1)
Git vroot status: weekly.2022.51-4-g21d6bd93
.git/config present: true
thirdparty/tcc status: thirdparty-macos-arm64 173c526e

**Comment by Delta456** [comment]
What are the values and types of `a`, `b` and `c`?

**Comment by spytheman** [comment]
@Delta456 something like this produces the same error:
```v
const a = 1

const b = 10

const c = a + b

fn can_be_compiled() {
	match 0 {
		a...b {}
		else {}
	}
}

fn cannot_be_compiled() {
	match 0 {
		a...c {}
		else {}
	}
}
```

**Comment by spytheman** [comment]
Note also that this:
```v
const a = 1

const b = 10

const c = a + b

dump(a)
dump(b)
dump(c)
```
produces: 
```
[zz.v:7] main.a: 1
[zz.v:8] main.b: 10
[zz.v:9] main.c: 11
```
and if you do `v -o x.c zz.v`, then: `grep _const_main__ x.c`:
```c
#define _const_main__a 1
#define _const_main__b 10
#define _const_main__c 11
         _v_dump_expr_int_literal(_SLIT("zz.v"), 7, _SLIT("main.a"), _const_main__a);
         _v_dump_expr_int_literal(_SLIT("zz.v"), 8, _SLIT("main.b"), _const_main__b);
         _v_dump_expr_int_literal(_SLIT("zz.v"), 9, _SLIT("main.c"), _const_main__c);
```
i.e. the value of `c` was known at compile time by V to be `11`, so in theory, it should have been known in the match range too, even though in practice, in the checker stage, where the error is produced, it is not yet known, because currently the const calculation happens I think in the transformer stage (run after the checker).

**Comment by spytheman** [comment]
@ytoml it is a tricky problem currently, given the current structure of the compiler. 
It may take some time to fix.

**Comment by spytheman** [comment]
One way to solve it, would be to turn off the checking for the range limits, for values that are not literals, or to delay that to cgen.
Another would be to do another simpler checking pass just for that, after the transform stage.
